spec/features/saml/saml_spec.rb

```diff
@@ -146,6 +146,16 @@
       expect(current_path).to eq(users_otp_path)
       expect(page).to have_content(I18n.t('devise.two_factor_authentication.otp_setup'))
     end
+
+    it 'adds acs_urls for current Rails env to CSP form_action' do
+      visit '/test/saml'
+      authenticate_user(user)
+
+      expect(page.response_headers['Content-Security-Policy']).
+        to include(
+          "form-action 'self' localhost:3000/test/saml/decode_assertion " \
+          'example.com/test/saml/decode_assertion;')
```

Comment:
Mixing the quotes style within the same string seems strange.  Took me a while to figure out what was quoted and what was not.